sync-core/src/main/java/com/cloudant/sync/query/UnindexedMatcher.java

```diff
@@ -233,17 +234,23 @@ private boolean executeSelectorTree(QueryNode node, DocumentRevision rev) {
                 operator = (String) operatorExpression.keySet().toArray()[0];
             }
 
-            Object expected = operatorExpression.get(operator);
+            Object expectedValues = operatorExpression.get(operator);
+            if (!(expectedValues instanceof List)) {
+                expectedValues = Arrays.asList(expectedValues);
+            }
```

Comment:
If you're considering this, might be worth doing it with `actual` too, so you're using a consistent pattern (and the `for` loop becomes simpler).